aspnetcore/data/ef-rp/intro/samples/StageSnapShots/cu-part3-sorting/Pages/Students/Edit.cshtml.cs

```diff
@@ -0,0 +1,65 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.AspNetCore.Mvc.RazorPages;
+using Microsoft.AspNetCore.Mvc.Rendering;
+using Microsoft.EntityFrameworkCore;
+using ContosoUniversity.Data;
+using ContosoUniversity.Models;
+
+namespace ContosoUniversity.Pages.Students
+{
+    public class EditModel : PageModel
+    {
+        private readonly ContosoUniversity.Data.SchoolContext _context;
+
+        public EditModel(ContosoUniversity.Data.SchoolContext context)
+        {
+            _context = context;
+        }
+
+        [BindProperty]
+        public Student Student { get; set; }
+
+        public async Task<IActionResult> OnGetAsync(int? id)
+        {
+            if (id == null)
+            {
+                return NotFound();
+            }
+
+            Student = await _context.Students.SingleOrDefaultAsync(m => m.ID == id);
+
+            if (Student == null)
+            {
+                return NotFound();
+            }
+            return Page();
+        }
+
+        public async Task<IActionResult> OnPostAsync(int? id)
+        {
+            if (!ModelState.IsValid)
+            {
+                return Page();
+            }
+
+            var studentToUpdate = await _context.Students.SingleOrDefaultAsync(
+                s => s.ID == id);
+
+            if (await TryUpdateModelAsync<Student>(
+                studentToUpdate,
+                "student",
+                s => s.FirstMidName, s => s.LastName, s => s.EnrollmentDate))
+            {
+                _context.Students.Update(studentToUpdate);
```

Comment:
With the pattern of query/set values/save there should be no need to call Update. Because EF is tracking the entity, it means if any property values are changes by TryUpdateModelAsync, then EF will detect those changes and send updates for the corresponding columns. Not only should Update not be needed, it will also cause all properties to be marked as modified whether they have changed or not, which means there will be more column updates sent to the database than are needed.